makeapp_osx.sh

```diff
@@ -1,93 +1,44 @@
-#!/bin/bash
-# Create a Mac OSX app bundle.
-# MUST be launched from the lifish directory
-# The lifish executable MUST already exist.
+#!/usr/bin/env bash
 
-if [[ $# < 1 ]]; then
-	echo "Usage: $0 <version>"
-	exit 1
-fi
+set -xe
 
-OUT_ZIP="lifish_BOOM_${1}_osx.zip"
-APPNAME="BOOM Remake"
-MACOS=(lifish)
-RESOURCES=(levels.json)
-ASSETS=(assets/fonts assets/graphics assets/sounds assets/screens assets/music assets/l10n)
-FRAMEWORK_PATH=/Library/Frameworks
-FRAMEWORKS=({FLAC,ogg,freetype,OpenAL,vorbis{,enc,file}}.framework)
-# Ugh...
-LOCAL_FRAMEWORK_PATH="../SFML-2.5.1-macos-clang/Frameworks"
-LOCAL_FRAMEWORKS=(sfml-{window,graphics,system,audio}.framework)
+sfml_version="2.6.1"
+sfml_filename="SFML-$sfml_version-sources.zip"
+sfml_url="https://www.sfml-dev.org/files/$sfml_filename"
 
-for i in ${MACOS[@]} ${RESOURCES[@]};  do
-	[[ -e $i ]] || {
-		echo "$i not found in this directory." >&2
-		exit 1
-	}
-done
+if [ ! -f "$sfml_filename" ]; then
+	curl -sSf -L -O "$sfml_url"
+	echo "5bf19e5c303516987f7f54d4ff1b208a0f9352ffa1cd55f992527016de0e8cb7  $sfml_filename" | shasum -a 256 -c
+	unzip "$sfml_filename"
+	mv "SFML-$sfml_version" SFML
+fi
 
-for i in ${FRAMEWORKS[@]}; do
-	[[ -d "$FRAMEWORK_PATH/$i" ]] || {
-		echo "$i not found in $FRAMEWORK_PATH". >&2
-		exit 1
-	}
-done
+pushd SFML
+# These arguments to patch avoid prompting to apply the changes repeatedly, making the script somewhat idempotent
+patch -N -r- -p 1 < ../osx/skip_obtaining_hid_devices.patch || true
+cmake -S . -B build -DCMAKE_INSTALL_PREFIX=install -DCMAKE_OSX_ARCHITECTURES="arm64;x86_64" -DSFML_BUILD_FRAMEWORKS=TRUE
+cmake --build build --config Release --target install -j "$(sysctl -n hw.logicalcpu)"
+popd
 
-# Lazy workaround to SFML not being *really* installed on my machine...
-for i in ${LOCAL_FRAMEWORKS[@]}; do
-	[[ -d "$LOCAL_FRAMEWORK_PATH/$i" ]] || {
-		echo "$i not found in $LOCAL_FRAMEWORK_PATH". >&2
-		exit 1
-	}
-done
+cmake -S . -B build -DRELEASE=true
+cmake --build build -j "$(sysctl -n hw.logicalcpu)"
 
-set -x
+sw_version="1.6"
```

Comment:
@silverweed Should this be 1.6.1? Note, you can create a release by running the workflow and downloading the artifact even if you don't have a Mac now.